### ISSUE TITLE
Terrible bugs (BB #710)

### ISSUE THREAD

**Issue by OregonAsari** [OregonCore/OregonCore#710]
This issue was migrated from bitbucket.
**Original Reporter:** Zurr
**Original Date:** 08.10.2014 17:06:04 GMT+0000
**Original Priority:** major
**Original Type:** bug
**Original State:** resolved
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/710

<hr>

Paladin seals,auras and blessings now stacks(you can use all of them together),
Hunter aspects stacks(you can use all of them together),
Hunter racial passive heroic presence and inspire presence - you can cancel them now like a buff by right clicking it in spell bar near portret,
Hunter spell tame beast - not working (invalid target),
Mages armor stacks,warlock curses stacks too

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** Zurr
**Original Date:** 08.10.2014 17:11:41 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/710/#comment-12852782

<hr>

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** LordUsagi
**Original Date:** 08.10.2014 17:27:25 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/710/#comment-12853067

<hr>

I can see the stacking issues being introduced in d7247bf4613d1b6c63a15f1f1dca4c90080f5e32
As for the draenie racials and tame beast would have been before that one eh?

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** Zurr
**Original Date:** 08.10.2014 17:29:11 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/710/#comment-12853099

<hr>

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** Zurr
**Original Date:** 08.10.2014 17:33:38 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/710/#comment-12853177

<hr>

yep, beast tame bug was in 2378 rev already
idk about racial s  i will check now

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** LordUsagi
**Original Date:** 08.10.2014 17:37:54 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/710/#comment-12853260

<hr>

Cheers, much appreciated.

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** Zurr
**Original Date:** 08.10.2014 18:10:26 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/710/#comment-12853881

<hr>

in 2378 rev draenei racial passives can be canceled

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** LordUsagi
**Original Date:** 08.10.2014 18:21:10 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/710/#comment-12854081

<hr>

Okay, Ive got mage armors and hunter aspects sorted. I need some assistance on paladin auras, seals and judgements. The way I believei t goes is

Judgements dont stack with Other Judgements
Blessings dont stack with Other blessings, including Greater
Seals dont stack with other Seals

Thats it

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** LordUsagi
**Original Date:** 08.10.2014 18:29:13 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/710/#comment-12854241

<hr>

```
#!sql

-- Fix up missing spell group stack rules
-- Thanks to Zear Forge for pointing out which spells were causing issues
DELETE FROM spell_group WHERE id IN (1008, 1009, 1012, 1013, 1014, 1015, 1016, 1017, 1018, 1019, 1020);
DELETE FROM spell_group WHERE spell_id IN (1044, 6940, -1016);
INSERT INTO spell_group (id, spell_id) VALUES
-- Mage
(1012, 7302),  -- Ice Armor
(1012, 30482), -- Molten Armor
(1012, 168),   -- Frost Armor
-- Hunter
(1013, 20043), -- Aspect of the Wild
(1013, 34074), -- Aspect of the Viper
(1013, 13161), -- Aspect of the Beast
(1013, 5118),  -- Aspect of the Cheetah
(1013, 13159), -- Aspect of the Pack
(1013, 13163), -- Aspect of the Monkey
(1013, 13165), -- Aspect of the Hawk
-- Warlock
(1014, 980),   -- Curse of Agony
(1014, 603),   -- Curse of Doom
(1014, 704),   -- Curse of Recklessness
(1014, 1490),  -- Curse of the Elements
(1014, 1714),  -- Curse of Tongues
(1014, 702),   -- Curse of Weakness
(1018, 28176), -- Fel Armor
(1018, 706),   -- Demon Armor
-- Paladin
-- Auras
(1015, 19876), -- Shadow Resistance Aura
(1015, 19891), -- Fire Resistance Aura
(1015, 19888), -- Frost Resistance Aura
(1015, 465),   -- Devotion Aura
(1015, 19746), -- Concentration Aura
(1015, 7294),  -- Retribution Aura
(1015, 32223), -- Crusader Aura
(1015, 20218), -- Sanctity Aura
-- Blessings, both greater and normal
(1009, 19977), -- Blessing of Light
(1009, 25890), -- Greater Blessing of Light
(1008, 1022),  -- Blessing of Protection
(1008, 23415), -- Improved Blessing of Protection
(1016, 1038),  -- Blessing of Salvation
(1016, 25895), -- Greater Blessing of Salvation
(1010, 1044),  -- Blessing of Freedom
(1010, 6940),  -- Blessing of Sacrifice
(1010, -1016), -- Salvation applied to blessings group
-- Seals
(1017, 20165), -- Seal of Light
(1017, 21084), -- Seal of Righteousness
(1017, 20166), -- Seal of Wisdom
(1017, 20164), -- Seal of Justice
(1017, 31892), -- Seal of Blood
(1017, 21082), -- Seal of the Crusader
-- Shamans
(1019, 324),   -- Lightning Shield
(1019, 24398), -- Water Shield
(1019, 974),   -- Earth Shield
-- Misc
-- Thori'dal & Quiver haste increases
(1020, 44972),
(1020, 29414),
(1020, 14829),
(1020, 14825),
(1020, 29416),
(1020, 29413),
(1020, 29417),
(1020, 29415),
(1020, 29418),
(1020, 14826),
(1020, 14828),
(1020, 14824),
(1020, 14827);

DELETE FROM spell_group_stack_rules WHERE group_id IN (1012, 1013, 1014, 1015, 1017, 1018, 1019, 1020);
INSERT INTO spell_group_stack_rules (group_id, stack_rule) VALUES
(1012, 1), -- Mage armors should not stack
(1013, 1), -- Hunter aspects should not stack
(1014, 2), -- Warlock curses should not stack for same caster
(1015, 2), -- Paladin Auras should not stack for same caster
(1017, 1), -- Paladin Seals should not stack
(1018, 1), -- Demon Armor and Fel Armor should not stack
(1019, 1), -- Shaman Shields should not stack
(1020, 1); -- Quiver bonuses + Thor'idal should not stack

```

Updated for paladin seals

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** Zurr
**Original Date:** 08.10.2014 18:35:52 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/710/#comment-12854342

<hr>

in last rev all paladins auras can be active at the same time for me 
but blessings and seals.... they can be active like 2 any blessings together and 2 any seals together
any 3 seal or blessing will "overwrite" last one and still 2 of them will be active

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** LordUsagi
**Original Date:** 08.10.2014 18:39:18 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/710/#comment-12854395

<hr>

Cheers, Ill sort it out when I get back from work in a couple of hours.
Could you test the code above and let me know how it goes. You can reload the table in game so no need for a restart.

As you can see we create a spell group, 1015 for all seals (rank 1 is fine, other ranks are calculated by core)
Then we assign a stack rule in the other table for group 1015, that those spells should only stack from different casters.

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** Zurr
**Original Date:** 08.10.2014 18:50:54 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/710/#comment-12854588

<hr>

ok i will check this now

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** Zurr
**Original Date:** 08.10.2014 19:07:07 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/710/#comment-12854907

<hr>

After fix only Hunter aspects is working properly now.
Other things still broken 
BUT  only  2 auras(paladin ) stacks now, 2 auras can be active only with Sanctity aura (Sanctity aura can be active with any other aura)
other auras dont stack now .  
Mage armor still bugged

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** Zurr
**Original Date:** 08.10.2014 19:13:48 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/710/#comment-12855017

<hr>

correction.
player can cast all blessings on self at the same time.
But "only" 3 seals can be active at the same moment on player,
if player cast fourth seal it will overwrite first seal

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** Zurr
**Original Date:** 08.10.2014 19:43:53 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/710/#comment-12855534

<hr>

Shaman shields and warlock armor bugged too

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** LordUsagi
**Original Date:** 09.10.2014 00:51:05 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/710/#comment-12860512

<hr>

I tested Paladin seals, mage armors, hunter aspects and curses. These all work fine. How are you testing them? Can I jump in game and take a look?

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** Zurr
**Original Date:** 09.10.2014 10:28:27 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/710/#comment-12869353

<hr>

curses now working fine,i was misinformed, and 1 paladin aura bugged http://www.wowwiki.com/Sanctity_Aura.
Shaman shields Lighting, earth and mana stacks together.
Fel Armor and Demon skin stacks together

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** LordUsagi
**Original Date:** 09.10.2014 12:54:05 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/710/#comment-12872202

<hr>

All right, comment updated with latest queries for Sanctity Aura, Shaman Shields (Lightning, Earth, Water - Let me know if I missed one), Fel Armor and Demon Armor now don't stack too.

Only thing that'll probably be important to fix next is Shaman totems stacking but, I'd need a lot of information for that.

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** Zurr
**Original Date:** 09.10.2014 13:08:09 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/710/#comment-12872487

<hr>

Mage armor. 
Frost Armor stacks with ice or molten )
I talked about this yesterday

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** Zurr
**Original Date:** 09.10.2014 13:33:23 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/710/#comment-12872978

<hr>

What about tame beast and racial passive spells
it can be fixed ? 
maybe some simple hack for a while?

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** LordUsagi
**Original Date:** 09.10.2014 13:36:39 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/710/#comment-12873023

<hr>

Simply don't right click the buff off? for the racial passives. I'll take a look at tame beast though.

UPDATE: Found the issue for tame beast, fix is on the way!

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** LordUsagi
**Original Date:** 09.10.2014 14:29:58 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/710/#comment-12874195

<hr>

[DB/SpellGroups] Add new stack rules for majority of spells that should/shouldn't stack.
                 Cleanup SQL for earlier commit.
                 Fix issue #478
                 Fix issue #710

→ <<cset 9822a2ca271c>>
